src/domain/parties/getPartiesByTypeAndID.js

```diff
@@ -0,0 +1,175 @@
+      // fspiop-proxy OR fspiop-source should be in the scheme
+      Logger.isErrorEnabled && Logger.error('Requester FSP not found')
+      throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Requester FSP not found')
+    }
+
+    const options = {
+      partyIdType: type,
+      partyIdentifier: params.ID,
+      ...(partySubIdOrType && { partySubIdOrType })
+    }
+
+    // see https://github.com/mojaloop/design-authority/issues/79
+    if (headers[Headers.FSPIOP.DESTINATION]) {
+      // the requester has specifid a destination routing header. We should respect that and forward the request directly to the destination
+      // without consulting any oracles.
+
+      // first check the destination is a valid participant
+      const destParticipantModel = await participant.validateParticipant(headers[Headers.FSPIOP.DESTINATION])
+      if (!destParticipantModel) {
+        // go to proxyCache, and try to get DESTINATION info (dfsp --> proxyId)
+        // if (!fount) throw error
+        // else        proxy to proxy
+
+        Logger.isErrorEnabled && Logger.error('Destination FSP not found')
+        throw ErrorHandler.Factory.createFSPIOPError(ErrorHandler.Enums.FSPIOPErrorCodes.ID_NOT_FOUND, 'Destination FSP not found')
+      }
+
+      // all ok, go ahead and forward the request
+      await participant.sendRequest(headers, headers[Headers.FSPIOP.DESTINATION], callbackEndpointType, RestMethods.GET, undefined, options, childSpan)
+      histTimerEnd({ success: true })
+      if (childSpan && !childSpan.isFinished) {
+        await childSpan.finish()
+      }
+      return
+    }
+
+    const response = await oracle.oracleRequest(headers, method, params, query, undefined, cache)
+    if (Array.isArray(response?.data?.partyList) && response.data.partyList.length > 0) {
+      // Oracle's API is a standard rest-style end-point Thus a GET /party on the oracle will return all participant-party records. We must filter the results based on the callbackEndpointType to make sure we remove records containing partySubIdOrType when we are in FSPIOP_CALLBACK_URL_PARTIES_GET mode:
+      let filteredResponsePartyList
+      switch (callbackEndpointType) {
+        case FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_GET:
+          filteredResponsePartyList = response.data.partyList.filter(party => party.partySubIdOrType == null) // Filter records that DON'T contain a partySubIdOrType
+          break
+        case FspEndpointTypes.FSPIOP_CALLBACK_URL_PARTIES_SUB_ID_GET:
+          filteredResponsePartyList = response.data.partyList.filter(party => party.partySubIdOrType === partySubIdOrType) // Filter records that match partySubIdOrType
+          break
+        default:
+          filteredResponsePartyList = response // Fallback to providing the standard list
+      }
+
+      if (filteredResponsePartyList == null || !(Array.isArray(filteredResponsePartyList) && filteredResponsePartyList.length > 0)) {
```

Comment:
The last part of this condition is not clear. `filteredResponsePartyList.length > 0` will never be true if `filteredResponsePartyList` is not an array, i.e. the condition checked before `!(Array.isArray(filteredResponsePartyList)`.

Reply:
it's not my impl., it's a legacy code:
https://github.com/mojaloop/account-lookup-service/blob/feat/fx-impl/src/domain/parties/parties.js#L113